lib/plaid/connector.rb

```diff
@@ -3,35 +3,102 @@
 require 'multi_json'
 
 module Plaid
-  # Private: A class encapsulating HTTP requests to the Plaid API servers
+  # Internal: A class encapsulating HTTP requests to the Plaid API servers
   class Connector
     attr_reader :uri, :http, :request, :response, :body
 
-    # Private: Prepare to run request.
+    # Internal: Default read timeout for HTTP calls.
+    DEFAULT_TIMEOUT = 120
+
+    # Internal: Prepare to run request.
     #
-    # path - The String path without leading slash. E.g. 'connect'
-    def initialize(path, id = nil)
+    # path    - The String path without leading slash. E.g. 'connect'
+    # subpath - The String subpath. E.g. 'get'
+    # auth - The Boolean flag indicating that client_id and secret should be
+    #        included into the request payload.
+    def initialize(path, subpath = nil, auth: false)
       unless Plaid.env
-        raise NotConfiguredError, "You must set Plaid.env before using any methods! Add a Plaid.config do .. end block somewhere in the initialization code of your program."
+        raise NotConfiguredError, 'You must set Plaid.env before using any ' \
```

Comment:
the three errors here seem to share slight variants on the same text, could we make a helper method? something like:

```

```